### PR TITLE
Orca: Add 2 NCF PyTorch examples with data_loader or XShards as inputs.

### PR DESCRIPTION
**Pull request by zpeng1898**
Add NCF pytorch examples : `train_data_loader.py` and `train_xshards.py` to the `NCF` directory, with a shared NCF-model `model.py`.

1.The `train_data_loader.py` example takes `data_loader` as the input of the model, supporting fitting the estimator with `ray` or `spark` backend: 

`# create the estimator`
`est = Estimator.from_torch(model=model_creator, optimizer=optimizer_creator,loss=loss_function, metrics=[Accuracy()],backend=Config["backend"])# backend="ray" or "spark"`
`# fit the estimator`
`est.fit(data=train_loader_func, epochs=1)`

2.The `train_xshards.py` example takes `XShards` as the input of the model, supporting fitting the estimator with `ray` or `spark` backend: 

`# create the estimator`
`est = Estimator.from_torch(model=model_creator, optimizer=optimizer_creator,loss=loss_function, metrics=[Accuracy()],backend=Config["backend"])# backend="ray" or "spark"`
`# fit the estimator`
`est.fit(data=train_shards, epochs=1,batch_size=Config["batch_size"],feature_cols=["x"],label_cols =["y"])`

### CODE REVIEW

File: python/orca/tutorial/pytorch/NCF/model.py [hunk 1]
```diff
@@ -60,15 +60,11 @@ def __init__(self, user_num, item_num, factor_num, num_layers,
...
-        output_modules = []
-        output_modules.append(nn.Linear(predict_size, 1))
-        output_modules.append(nn.Sigmoid())
-        self.predict_layer = nn.Sequential(*output_modules)
+        self.predict_layer = nn.Linear(predict_size, 1)
 
         self._init_weight_()
 
     def _init_weight_(self):
-        """ We leave the weights initialization here. """
```
Comment (hkvision) [inline python/orca/tutorial/pytorch/NCF/model.py:71]: keep this comment?

File: python/orca/tutorial/pytorch/NCF/train_data_loader.py [hunk 2]
```diff
@@ -40,104 +40,98 @@
...
-    data_X["label"] = labels
-
     # train test split
-    data_X = data_X.values.tolist()
     _, test_data = train_test_split(data_X, test_size=0.2, random_state=100)
-    test_data = pd.DataFrame(test_data, columns=["user", "item", "label"], dtype=np.int64)
-    test_data["label"] = test_data["label"].astype(np.float)
 
-    test_dataset = NCFData(test_data)
+    test_dataset = NCFData(test_data, item_num, train_mat, config['num_ng'], True)
```
Comment (hkvision) [inline python/orca/tutorial/pytorch/NCF/train_data_loader.py:133]: change shuffle to be False for test?

File: python/orca/tutorial/pytorch/NCF/train_data_loader.py [hunk 3]
```diff
@@ -146,35 +140,47 @@ def test_loader_func(config, batch_size):
...
     model.train()
     return model
 
 
 def optimizer_creator(model, config):
-    return optim.Adam(model.parameters(), lr=0.001)
+    return optim.Adam(model.parameters(), lr=config['lr'])
 
-loss = nn.BCEWithLogitsLoss()
+loss_function = nn.BCEWithLogitsLoss()
```
Comment (hkvision) [inline python/orca/tutorial/pytorch/NCF/train_data_loader.py:161]: just loss?

File: python/orca/tutorial/pytorch/NCF/train_xshards.py [hunk 4]
```diff
@@ -39,18 +39,8 @@
...
-        usecols=[0, 1], dtype={0: np.int64, 1: np.int64})
-    data_X = data_X.partition_by("user")
-
-    user_set = set(data_X["user"].unique())
-    item_set = set(data_X["item"].unique())
-    user_num = max(user_set) + 1
-    item_num = max(item_set) + 1
-    return data_X, user_num, item_num
+dataset_dir = "./ml-1m"
+num_ng = 4
```
Comment (hkvision) [inline python/orca/tutorial/pytorch/NCF/train_xshards.py:43]: put num_ng in ng_sampling?

File: python/orca/tutorial/pytorch/NCF/train_xshards.py [hunk 5]
```diff
@@ -105,30 +95,40 @@ def split_dataset(data):
...
     model.train()
     return model
 
 
 def optimizer_creator(model, config):
-    return optim.Adam(model.parameters(), lr=0.001)
+    return optim.Adam(model.parameters(), lr=config['lr'])
 
-loss = nn.BCEWithLogitsLoss()
+loss_function = nn.BCEWithLogitsLoss()
```
Comment (hkvision) [inline python/orca/tutorial/pytorch/NCF/train_xshards.py:109]: same as above

File: python/orca/tutorial/pytorch/NCF/train_data_loader.py [hunk 6]
```diff
@@ -40,104 +40,98 @@
...
+        user = features[idx][0]
+        item = features[idx][1]
+        label = float(labels[idx])
         return user, item, label
 
 
 def train_loader_func(config, batch_size):
     data_X = pd.read_csv(
-        "ml-1m/ratings.dat",
+        config['dataset_dir']+"/ratings.dat",
```
Comment (hkvision) [inline python/orca/tutorial/pytorch/NCF/train_data_loader.py:89]: change to os.path.join?
